org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterTuyaButton.java

```diff
@@ -0,0 +1,216 @@
+            logger.error("{}: Error opening device client controls", endpoint.getIeeeAddress());
+            return false;
+        }
+
+        if (serverCluster == null) {
+            logger.error("{}: Error opening device server controls", endpoint.getIeeeAddress());
+            return false;
+        }
+
+        clientCluster.addCommandListener(this);
+        serverCluster.addAttributeListener(this);
+
+        // Add Tuya-specific command
+        //
+        HashMap<Integer, Class<? extends ZclCommand>> commandMap = new HashMap<>();
+        commandMap.put(TuyaButtonPressCommand.COMMAND_ID, TuyaButtonPressCommand.class);
+        clientCluster.addClientCommands(commandMap);
+
+        return true;
+    }
+
+    @Override
+    public void disposeConverter() {
+        if(clientCluster != null) {
+            clientCluster.removeCommandListener(this);
+        }
+        if (serverCluster != null) {
+            serverCluster.removeAttributeListener(this);
+        }
+    }
+
+    @Override
+    public void handleRefresh() {
+        // nothing to do, as we only listen to commands
+    }
+
+    @Override
+    public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
+        // This converter is used only for channels specified in static thing types, and cannot be used to construct
+        // channels based on an endpoint alone.
+        return null;
+    }
+
+    @Override
+    public boolean commandReceived(ZclCommand command) {
+        logger.debug("{} received command {}", endpoint.getIeeeAddress(), command);
+        Integer thisTxId = command.getTransactionId();
+        if(lastTxId == thisTxId)
+        {
+            logger.debug("{} ignoring duplicate command {}", endpoint.getIeeeAddress(), thisTxId);
+        }
+        else if (command instanceof TuyaButtonPressCommand) {
+            TuyaButtonPressCommand tuyaButtonPressCommand = (TuyaButtonPressCommand) command;
+           thing.triggerChannel(channel.getUID(), getEventType(tuyaButtonPressCommand.getPressType()));
+           clientCluster.sendDefaultResponse(command, ZclStatus.SUCCESS);
+        }
+        else {
+           logger.warn("{} received unknown command {}", endpoint.getIeeeAddress(), command);
+        }
+
+        lastTxId = thisTxId;
+        return true;
+    }
+
+    @Override
+    public void attributeUpdated(ZclAttribute attribute, Object val) {
+        logger.debug("{}: ZigBee attribute reports {}", endpoint.getIeeeAddress(), attribute);
+    }
+
+    private String getEventType(Integer pressType)
+    {
+        switch(pressType)
+        {
+            case 0:
+                return CommonTriggerEvents.SHORT_PRESSED;
+            case 1:
+                return CommonTriggerEvents.DOUBLE_PRESSED;
+            case 2:
+                return CommonTriggerEvents.LONG_PRESSED;
+            default:
+                logger.warn("{} received unknown pressType {}", endpoint.getIeeeAddress(), pressType);
```

Comment:
Being petty, but please addd a `:` after the `{}` for the address - it's good to keep things standard as sometimes processing of logs looks for certain features.